### PR TITLE
[skip ci] Update CONTRIBUTING.md with pre-commit info

### PR DESCRIPTION
**Pull request by blozano-tt**
### Problem description
We now heavily use pre-commit, but don't give any background on it, or instructions on how to set it up.

### What's changed
Added instructions.

cc: @bbeggsTT

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -429,6 +429,43 @@ cat generated/watcher/watcher.log  # See k_ids field for each core in the last d
...
+For more details on pre-commit, you can visit the [official documentation](https://pre-commit.com/).
+
+### How to Set Up Pre-commit Locally
+
+To set up pre-commit on your local machine, follow these steps:
+
+1. **Install Pre-commit**:
+   Ensure you have Python installed, then run:
+   ```bash
+   pip install pre-commit
```
Comment (tt-rkim) [inline CONTRIBUTING.md:456]: We have this in the python_env. Should we consider that while making these instructions?

**Review by tt-rkim — APPROVED** [review]
Nit pick about python env. LGTM otherwise

File: CONTRIBUTING.md (hunk 1 above)
Comment (blozano-tt) [reply CONTRIBUTING.md:456]: Interesting. I have never used the python_env. I typically bring in pre-commit via Anaconda or homebrew. 

Given that its not mandatory to use the python_env for this, do we need to guide people that direction?

Comment (blozano-tt) [reply CONTRIBUTING.md:456]: I'll add a note

**Review by bbeggsTT — COMMENTED** [review]
Formatting and language looks fine.

**Pull request merged**